### ISSUE TITLE
Publish profile to vault

### ISSUE THREAD

**Issue by juliangruber** [hypergraph-xyz/desktop#180]
Publish the profile to the vault, if the user opts in.

- [x] On the last welcome screen, ask user to opt in to Vault
- [x] Store preference
- [x] Allow user to opt out via menu bar entry
- [x] Submit the unversioned profile link to the vault

**Comment by juliangruber** [comment]
@psobrakseaton the design needed for this is an update to the last modal that lets the user opt in to continuously publishing their work (profile and contents) to the vault.

The screen currently looks like this:

![image](https://user-images.githubusercontent.com/10247/89156002-3d4e1b00-d56a-11ea-8252-704e3544826c.png)

What do you think about adding a second button for also opting in?

@jameslibscie we also need updated copy that says that you can opt in to hosting your profile and registered content on the vault for free (for now). @chartgerink might be able to give some context. Matching button labels would be great as well.

**Comment by psobrakseaton** [comment]
> What do you think about adding a second button for also opting in?

I think having two buttons is a good way to go. Two buttons with equal hierarchy that say "Store my work in the vault" and a "Skip for now" perhaps?

We could also have a checkbox or a toggle, but that would require the creation of those elements and would take a while longer.

**Comment by psobrakseaton** [comment]
[Here is the Figma file for this issue.](https://www.figma.com/file/gbDl3zCgqiD8woAMe6JvvZ/180-Publish-profile-to-vault?node-id=0%3A1)

I'll update here on GH when the designs are ready to go.

**Comment by chartgerink** [comment]
@jameslibscie, @lhehnke and I are working on the copy for this screen. Feel free to implement for now and we can PR in the edits as part of the welcome modal work item, I'd say?

@psobrakseaton the design looks nice 👍

**Comment by juliangruber** [comment]
@psobrakseaton this screen needs a loading state. With the single-button design, the button would change into a loading animation. What should happen when the link is clicked?

**Comment by psobrakseaton** [comment]
> @psobrakseaton this screen needs a loading state. With the single-button design, the button would change into a loading animation. What should happen when the link is clicked?

Yup! We also need a way to indicate that the vault is enabled, and we need some way after the fact to turn on the vault if they decide to skip it for now.

When I posted the link earlier, it wasn't a finished design, just marking that we have a Figma file for this issue.

**Comment by chartgerink** [comment]
Is it an idea to make these actions part of a "Vault" menu bar item to minimise design work needed right now? @psobrakseaton might be efficient to do that for now and not have to design out all these states in the next few weeks.

**Comment by psobrakseaton** [comment]
> Is it an idea to make these actions part of a "Vault" menu bar item to minimise design work needed right now? @psobrakseaton might be efficient to do that for now and not have to design out all these states in the next few weeks.

I think this is a great idea. We should have some discussions about what items we need in the menu, but I am definitely in favor of that.

**Comment by chartgerink** [comment]
@psobrakseaton can you finalise the designs? This is the copy. I will make an issue for the menu bar so no need to think about this here.

## Introducing the Vault

We use peer-to-peer technology: Hypergraph is powered by the people using it! Share your content as you'd like, without anybody getting in between.

Hypergraph Vault makes your work available even when your computer is offline. Plus, we're working with libraries on archiving it for future generations! 👵👨👶

Until January 1st, 2021, Hypergraph Vault is free, while we figure out the costs. We promise to keep it to one-time fees for lifetime service.

<Enable vault> <Skip for now...>

**Comment by psobrakseaton** [comment]
> @psobrakseaton can you finalise the designs? This is the copy. I will make an issue for the menu bar so no need to think about this here.
> 
> ## Introducing the Vault
> We use peer-to-peer technology: Hypergraph is powered by the people using it! Share your content as you'd like, without anybody getting in between.
> 
> Hypergraph Vault makes your work available even when your computer is offline. Plus, we're working with libraries on archiving it for future generations! 👵👨👶
> 
> Until January 1st, 2021, Hypergraph Vault is free, while we figure out the costs. We promise to keep it to one-time fees for lifetime service.

Fantastic! This should make the process very quick! I will get on this ASAP. Thanks!

**Comment by psobrakseaton** [comment]
Do you think it is ok to have a simple button an no sort of explicit checkmark? 

<img width="717" alt="Screen Shot 2020-08-19 at 13 38 20" src="https://user-images.githubusercontent.com/28573875/90630561-8b794480-e221-11ea-8ec0-2663a57c89ed.png">

**Comment by juliangruber** [comment]
@psobrakseaton what's the current state of this?

**Comment by psobrakseaton** [comment]
The interaction on this has been drastically simplified since discussing it with @juliangruber and @jameslibscie, I previously had several states and a a success indicator and the like, now it has been reduced to just this:

The design is identical to what I posted before.

- [ ] This modal goes before the final Liberate Science promo modal
- [ ] The "Enable the Vault" button will advance to the next modal.
- [ ] The same interaction for the "Skip for now..." as the enabling button.
- [ ] The only design change for the promo modal is the removal of the back arrow.

I played around with some difference designs for the promo modal, but that is all out of scope. Let's go with this simple design for now.
